src/gg.regression.unity.bots/Runtime/Scripts/RGTestUtils.cs

```diff
@@ -41,7 +41,21 @@ public static IEnumerator WaitForScene(string sceneName, int timeout = 5)
             yield return null;
             Assert.IsTrue(loaded, $"Scene {sceneName} failed to load within {timeout} seconds");
         }
-
+        
+        /// <summary>
+        /// Wait for the specified condition to become true with a timeout.
+        /// This is similar to WaitUntil(), but also includes a failing assertion if the timeout expires.
+        /// </summary>
+        /// <param name="condition">The condition that should become true</param>
+        /// <param name="timeout">Maximum time to wait for the condition to become true (in seconds)</param>
+        public static IEnumerator WaitUntil(Func<bool> condition, int timeout = 5)
```

Comment:
Not related to this method, but having TestUtils in our Runtime folder of our SDK feels off

Reply:
Maybe we can name this differently - this is more of an artifact of our SDK being a testing framework, so this is not a utility for our tests, but rather for users of our SDK to use in their tests. Happy to rename this!

Reply:
Could just be RGUtils

Reply:
We already have at least 1 `RGUtils` class elsewhere in the code, so be careful with naming here

Reply:
Do we want to move the file out into its own assembly, e.g. `RegressionGames.TestFramework`? This is similar to what the Input System does for its own [test framework](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/manual/Testing.html).

Reply:
@RG-nAmKcAz @vontell What would be your preferred approach for resolving this?
Options I can see from the above discussion:
1) rename to RGUtils 
2) move into its own test assembly
3) leave as is

Reply:
1 . Would need to be something other than RGUtils as mentioned above
2. Is fine

For simplicity I would probably go with #1 but just pick some other naming

Reply:
I will go with option 2 (moving into its own test assembly), since if we want to have utilities for others to use in Unity Test Runner then having it in its own test framework assembly will also allow us to access/extend the NUnit APIs in the future if desired (these are normally unavailable in the Runtime assembly).

However, we have various places that depend on this class ([SDK docs](https://docs.regression.gg/gameplay-sessions/gameplay-sessions-reference) and  [BossRoom's RecordingTest](https://github.com/Regression-Games/RGBossRoom/blob/main/Assets/Tests/Runtime/RecordingTest.cs)), so I've created a ticket REG-1888 for this and I think it would be better to have this be a separate set of PRs that addresses all these at once.